File: _experimental/_julia/numbers/test/numbers_recursive_test.jl
#!/usr/bin/julia

using Test
include("../src/NumbersRecursive.jl")
using .NumbersRecursive

@testset "Recursive Tests" begin
  @testset "Fibonacci Tests" begin
    @test fibonacci(5) == 5
  end

  @testset "Factorial Tests" begin
    @test factorial_rec(5) == 120
  end

  @testset "Sum Numbers Tests" begin
    @test sum_numbers(5) == 15
  end
end

File: _experimental/_zig/numbers/test/numbers_recursive_test.zig
const std = @import("std");
const numbers_recursive = @import("numbers_recursive");

test "recursive Fibonacci" {
    try std.testing.expect(numbers_recursive.fibonacci(5) == 5);
}

test "recursive Factorial" {
    try std.testing.expect(numbers_recursive.factorial(5) == 120);
}

test "recursive Sum of N" {
    try std.testing.expect(numbers_recursive.sum_numbers(5) == 15);
}
